frontend/app/screens/goal-detail-screen/goal-detail-screen.tsx

```diff
@@ -219,13 +231,13 @@ export const GoalDetailScreen = observer(function GoalDetailScreen() {
   const sunday = [];
 
   for (const goal of STgoals) {
-    for (let i = 0; i < goal.mon.length; i++) monday.push([goal.mon[i], goal.title, goal.id]);
-    for (let i = 0; i < goal.tue.length; i++) tuesday.push([goal.tue[i], goal.title, goal.id]);
-    for (let i = 0; i < goal.wed.length; i++) wednesday.push([goal.wed[i], goal.title, goal.id]);
-    for (let i = 0; i < goal.thu.length; i++) thursday.push([goal.thu[i], goal.title, goal.id]);
-    for (let i = 0; i < goal.fri.length; i++) friday.push([goal.fri[i], goal.title, goal.id]);
-    for (let i = 0; i < goal.sat.length; i++) saturday.push([goal.sat[i], goal.title, goal.id]);
-    for (let i = 0; i < goal.sun.length; i++) sunday.push([goal.sun[i], goal.title, goal.id]);
+    if (goal.mon) for (let i = 0; i < goal.mon.length; i++) monday.push([goal.mon[i], goal.title, goal.id]);
```

Comment:
Do we need to create another PR for this change? Is this if statement needed?

Reply:
i got an error when going from common-goals to the detail screen, it's there for safety reasons. do you think we should just edit the common goals to evade this error or keep it for code safety?